lib/pure/algorithm.nim

```diff
@@ -8,19 +8,40 @@
 #
 
 ## This module implements some common generic algorithms.
+##
+## .. code-block::
+##    import algorithm
+##
+##    var birth = ["2000 John", "1995 Marie", "2010 Jane"]
```

Comment:
If you turn this into `seq[tuple[year: int, name: string]]` it might be easier to showcase the difference between `sort` with the default `system.cmp` and a `sort` where you defined your own `cmp`, something like this:

```nim
import algorithm

type People = tuple
  year: int
  name: string

var a: seq[People]

a.add((2000, "John"))
a.add((2005, "Marie"))
a.add((2010, "Jane"))

a.sort()
assert a == @[(year: 2000, name: "John"), (year: 2005, name: "Marie"), (year: 2010, name: "Jane")]

proc myCmp(x, y: People): int =
  if x.name < y.name: -1 else: 1

a.sort(myCmp)
assert a == @[(year: 2010, name: "Jane"), (year: 2000, name: "John"), (year: 2005, name: "Marie")]
```

(You might want to reuse some of these ideas for later, when you describe those functions allowing for a custom `cmp`)